### ISSUE TITLE
🚨 REVERT Warp AI montly request limit

### ISSUE THREAD

**Issue by Flecart** [warpdotdev/Warp#4829]
### Describe the solution you'd like?

**Bring back** the 100 requests daily! At least use daily requests!

I am amazed that nobody is talking about this. I liked the integrated AI chat quite a lot. But 40 requests per month are useless.
So you can either do these two things:

1. Allow personal API Keys and endpoints setup
2. Allow local models chat
3. Bring back the daily request limit, you can lower it, but not from 100 to 1, as we have it now.

Warp is nothing without the community, and we can pressure to change this. I don't want to pay teams version for AI usage.

**Vote** this issue in order to be heard from warp.

### How important is this feature to you?

5 (Can't work without it!)

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

Ignore

**Comment by dannyneira** [comment]
Thanks for this request @Flecart. We value your feedback on the changes to AI limits.

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by Bryce-Colton** [comment]
I didn't use Warp yet. But I can confirm that 40 requests a month worth nothing. Even the 200 requests per month is useless to pretend this ai integration is a killer feature. 30 requests per day is a start imho (with costs obviously).

I think the allowing third party ia request would be a fair move. Like jetbrains, they provide their ai integration but allow users to use copilot plugin too.

**Comment by Flecart** [comment]
> I didn't use Warp yet. But I can confirm that 40 requests a month worth nothing. Even the 200 requests per month is useless to pretend this ai integration is a killer feature. 30 requests per day is a start imho (with costs obviously).
> 
> I think the allowing third party ia request would be a fair move. Like jetbrains, they provide their ai integration but allow users to use copilot plugin too.

Before the update you had 100 requests daily for free. It's a big change from 100 daily to about than 1 daily on average. 
But agree with the 30 and the third party AI thing.

**Comment by Flecart** [comment]
Check #2788 for discussions about third party.

**Comment by velik98** [comment]
I just noticed that! It is a big downgrade. 100 requests a day to 40 a month?  That's a huge difference. Honestly warp AI is one of the most used features and now I just found out I'm half thru my monthly limit. Please bring back the original limit. Even if it would be half like 50 requests per day it  is better then what you came with. This is almost unusable.

**Comment by Flecart** [comment]
> This change makes the terminal completely un-useable. I don't mind putting in my own key if that's what it takes.. per month is... grating.. I can't see that message constantly for another 21 days and not switch terminals lol x-x

Agree, I've dropped warp because of this change.

**Comment by Patcybermind** [comment]
yeah 40 a month is pointless and the 22 dollars a month is way too much i wouldnt even consider it if it were 5 dollars after its just a terminal

**Comment by tnypxl** [comment]
Every AI-supported app I use right now thinks it needs $20+ a month from me to access their AI features when we all know how and to whom these API calls are being made.

I really enjoy using Warp. But locking AI features behind a tier is kind of wild.

**Comment by ismaslov** [comment]
The recent limit on free queries in your terminal has been very frustrating. Now, I find myself disappointed for 25 days each month. Please reconsider this restriction.

**Comment by malleybu** [comment]
I've only just installed Warp AI and I'ver just been playign around with it, testing, and didn't relize it was limited request, I've run out.... oh-well, I have a look again in 29days.

**Comment by PerimeterViktor** [comment]
Agree, the limit is pretty low. 5$/month for ability to bring own key would be great, 22 is steep

**Comment by suchcodemuchwow** [comment]
Moving to [Alacritty](https://github.com/alacritty/alacritty) thanks to your _"decisions"_

If I pay $25 **it only saves me 1 click + copy paste** compared to using chatgpt free. And paid version of chatGPT is $20 dollar with many other use cases than just command suggestion.

I don't think this is anymore "by developers for developers" mindset. There was only 2 reasons i was using warp over others: ai and text editing/bottom input. I would suggest you charge text editing for $15 or more as well. 

Goodluck with your journey.

**Comment by PRESIDENT810** [comment]
Agree, at least could you please open an API for local deployed LLMs? Now we have ollama and tons of inference engine (vllm, llama.cpp), it would be easy for devs to deploy their on model and connect to an API

**Comment by V1067** [comment]
The only thing that would make me give up Warp is Warp AI.

**Comment by scmrus** [comment]
The limit of 40 requests per month is a joke. It should be a daily limit in order to be more useful. 

I'm considering switching to https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-in-the-command-line or similar tools that use local models or have direct integration to the API (even https://www.raycast.com/pro costs $8 or free if you are using existing API key and plugin).

I really like Warp, but assuming that developers are incapable of integrating with alternatives or using the API directly is short-sighted. I understand the need for monetization, but it feels like an attempt to oversell overpriced API integration and seems misleading.

**Comment by NOSHIRTEDKEEPER** [comment]
Just add support for llama, then it doesn't cost you anything

**Comment by Abstracting** [comment]
Maybe there could be a feature in the pro / team plan that you can "bring you own api key" in addition to get around the limit per month? I just think that the psychological barrier "Dont use up your requests for simple stuff" is a bit counterproductive.

**Comment by Code4Bug** [comment]
When will local Ollama support be available?

**Comment by Dnonmi** [comment]
API keys might take a while since most models have different function calls, they should choose the one which is more open to support first (right now it's Google by a long mile).
I second that we want both methods available, local is good too but people are not considering their energy bills at all it seems. Running a server like LM Studio or ollama all the time, will be more expensive than what you were willing to pay for AI services, so be mindful of that.

**Comment by feribg** [comment]
I think people will slowly move away from tools locking them into a wrapper of openai or anthropic into the equivalent ones allowing them control over the AI vendor and setup. The same backlash is already way underway for cursor for instance so i'm assuming it's a matter of time before something gives. Noone wants to be upcharged for a hotkey and a openai wrapper...

It's ridiculous to not allow local models and common routers and just charge a markup to the ai vendor.

here's a trivial solution https://github.com/AnswerDotAI/shell_sage

**Comment by orckid-lab** [comment]
$15 - 1000 requests per month = 33 requests per day
$40 - 3000 requests per month = 100 requests per day

unie unlimited at $20 = 1,500-2000 per day.

Issue not addressed since 2024.

**Comment by 09deepesh** [comment]
i Was working on a project and the limit exceeded at important time  😶😶

**Comment by agent-carvajal** [comment]
time to uninstall warp, happy to see everyone realized is a scam
